addon/components/file-browser/component.js

```diff
@@ -0,0 +1,23 @@
+import Ember from 'ember';
+import layout from './template';
+
+/* use:
+ * {{file-browser rootNode=node selectedFile=file
+ *   onClickFile=(action) onClickNode=(action)}}
+ */
+export default Ember.Component.extend({
+    layout,
+    classNames: ['file-browser'],
+
+    selectedPath: Ember.computed('selectedFile', function() {
+        let file = this.get('selectedFile');
+        if (!file) {
+            return null;
+        }
+        let pathArray = this.get('selectedFile.materializedPath').split('/');
```

Comment:
If you click on a provider like osfstorage in the file browser, it fails because the context is the the API response from `/v2/nodes/<node_id>/files/`.  If we're mimicking the file detail page, the provider is not a link.
```
{
    "data": [
        {
            "relationships": {
                "files": {
                    "links": {
                        "related": {
                            "href": "http://localhost:8000/v2/nodes/3e27w/files/osfstorage/",
                            "meta": {}
                        }
                    }
                }
            },
            "links": {
                "upload": "http://localhost:7777/v1/resources/3e27w/providers/osfstorage/",
                "new_folder": "http://localhost:7777/v1/resources/3e27w/providers/osfstorage/?kind=folder"
            },
            "attributes": {
                "node": "3e27w",
                "path": "/",
                "kind": "folder",
                "name": "osfstorage",
                "provider": "osfstorage"
            },
            "type": "files",
            "id": "3e27w:osfstorage"
        }
    ],
    "links": {
        "first": null,
        "last": null,
        "prev": null,
        "next": null,
        "meta": {
            "total": 1,
            "per_page": 10
        }
    }
}
```